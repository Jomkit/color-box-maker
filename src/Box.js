import React from 'react'
import "./Box.css";

const Box = ({width, height, backgroundColor, deleteBox}) => {
  return (
    <div className='Box' style={ {width: width, height: height, backgroundColor: backgroundColor,} }>
        <button className='Box-deleteBtn' onClick={deleteBox}>X</button>
    </div>
  )
}

export default Box;