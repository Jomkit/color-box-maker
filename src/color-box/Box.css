.Box {
    display: inline-block;
    text-align: start;
    border-radius: 10px;
}

.Box-deleteBtn {
    margin: 0;
    border: none;
    background: none;
}

.Box-deleteBtn:hover {
    cursor: pointer;
    border: 1px solid black;
    border-radius: 5px;
}