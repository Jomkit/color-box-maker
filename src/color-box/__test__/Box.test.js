import { render, fireEvent } from "@testing-library/react";
import Box from "../Box";

const mockDelete = jest.fn();

//Smoke Test
it('should render without crashing', () => { 
    render(<Box key={1} width="200px" height="200px" backgroundColor="blue" deleteBox />);
})

// Snapshot Test
it('should matchshot', () => { 
    const { asFragment } = render(<Box key={1} width="200px" height="200px" backgroundColor="blue" />);

    expect(asFragment()).toMatchSnapshot();
})  

// Test that function is called
// 6/7/24 - Why does this delete the box even if the function isn't passed as prop?
it('should call deleteBox when delete button clicked', () => { 
    const { queryByText } = render(<Box key={1} width="200px" height="200px" backgroundColor="blue" deleteBox={mockDelete} />);

    const deleteBtn = queryByText('X');
    expect(deleteBtn).toBeInTheDocument();
    fireEvent.click(deleteBtn);
    expect(mockDelete).toHaveBeenCalledTimes(1);
})