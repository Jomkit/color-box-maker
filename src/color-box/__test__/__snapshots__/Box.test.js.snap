// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should matchshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
    style="width: 200px; height: 200px; background-color: blue;"
  >
    <button
      class="Box-deleteBtn"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
